xwiki-platform-core/xwiki-platform-ratings/xwiki-platform-ratings-ui/src/main/resources/XWiki/Ratings.xml

```diff
@@ -446,20 +446,22 @@ position: relative;
     </class>
     <property>
       <content>{{velocity}}
-#set($canView = $services.security.authorization.hasAccess("view", "XWiki.RatingsMacros"))
+#set($canView = $services.security.authorization.hasAccess('view', 'XWiki.RatingsMacros'))
 #if($canView)
   {{include reference='XWiki.RatingsMacros'/}}
-  #set($configurationClass = 'XWiki.RatingsConfigClass')
+#end
+{{/velocity}}
```

Comment:
It's not intuitive, as I also suggested to @evalica initially to group them, but the order of rendering is important and affected by the 2 approaches.

If they are all inside a single velocity macro, the velocity code executes first, when there is no `displayFullRating` defined and will print it as it is. After the velocity execution, the wiki syntax will be rendered and the `include` macro will be executed and it will define the `displayFullRating` macro, but it is too late by now.

If you separate them, the first velocity macro block renders completely and also includes and defines the `displayFullRating` macro, and, as a result, the separate velocity macro block`s rendering will now properly have the macro defined and can execute it when the velocity execution is done.

All of this is the result of the condition that is set on the `include` wiki macro call, which probably aims to improve performance by not defining useless macros (and whatever else is in that page).

Perhaps another option would have been to group them and, instead of the include wiki macro call, to use the old velocity `#includeMacros` macro, or something along those lines.